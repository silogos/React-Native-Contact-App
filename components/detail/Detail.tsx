import React from "react";
import { StyleSheet } from "react-native";
import { ThemedView } from "@/components/ThemedView";
import { useThemeColor } from "@/hooks/useThemeColor";
import { ThemedText } from "../ThemedText";

type DetailType = {
  label: String;
  value: String;
};

export default function Detail({ label, value }: DetailType) {
  const color = useThemeColor({}, "border");
  return (
    <ThemedView
      colorName="card"
      style={{ ...styles.container, borderBlockColor: color }}
    >
      <ThemedText type="defaultSemiBold" colorName="foreground">
        {label}:
      </ThemedText>
      <ThemedText type="defaultSemiBold" colorName="primary">
        {value}
      </ThemedText>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    paddingHorizontal: 18,
    paddingVertical: 5,
    borderBottomWidth: 1,
    borderTopWidth: 1,
  },
});
