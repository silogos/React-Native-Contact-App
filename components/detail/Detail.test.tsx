import React from "react";
import { render } from "@testing-library/react-native";

import Detail from "./Detail";

describe("Detail", () => {
  it("renders correctly", () => {
    const elm = render(<Detail label={"title"} value={"value"} />).toJSON();

    expect(elm).toMatchSnapshot();
  });

  it("Render with data", () => {
    const label = "Test Title";
    const val = "Test Title";
    const { getByText } = render(<Detail label={label} value={val} />);

    expect(getByText(label)).toBeTruthy();
    expect(getByText(val)).toBeTruthy();
  });
});
