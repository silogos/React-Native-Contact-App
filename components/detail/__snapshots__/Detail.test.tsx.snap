// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "borderBlockColor": "#e4e4e7",
        "borderBottomWidth": 1,
        "borderTopWidth": 1,
        "flexDirection": "column",
        "paddingHorizontal": 18,
        "paddingVertical": 5,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#09090b",
        },
        undefined,
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "600",
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    title
    :
  </Text>
  <Text
    style={
      [
        {
          "color": "#20A4DC",
        },
        undefined,
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "600",
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    value
  </Text>
</View>
`;
